crates/ruma-macros/src/events/event_enum.rs

```diff
@@ -419,10 +421,51 @@ fn expand_content_enum(
     })
 }
 
+/// Create a full content enum from `EventEnumInput`.
+fn expand_full_content_enum(
+    kind: EventKind,
+    events: &[EventEnumEntry],
+    docs: &[TokenStream],
+    attrs: &[Attribute],
+    variants: &[EventEnumVariant],
+    ruma_common: &TokenStream,
+) -> syn::Result<TokenStream> {
+    let ident = kind.to_full_content_enum();
+
+    let content: Vec<_> = events
+        .iter()
+        .map(|event| {
+            let stable_name = event.stable_name()?;
+            Ok(to_event_content_path(kind, stable_name, &event.ev_path, None))
+        })
+        .collect::<syn::Result<_>>()?;
+
+    let variant_decls = variants.iter().map(|v| v.decl()).collect::<Vec<_>>();
+
+    Ok(quote! {
+        #( #attrs )*
+        #[derive(Clone, Debug)]
+        #[allow(clippy::large_enum_variant)]
+        #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+        pub enum #ident {
+            #(
+                #docs
+                #variant_decls(#ruma_common::events::FullStateEventContent<#content>),
+            )*
+            #[doc(hidden)]
+            _Custom {
+                event_type: crate::PrivOwnedStr,
+                redacted: bool,
```

Comment:
I guess this exists for debug purposes?

Reply:
Yes, I'm not sure is necessary, but I'm thinking "why not?".